sound/soc/intel/common/soc-acpi-intel-match-common.c

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * soc-apci-intel-match-common.c - helper functions for ACPI enumeration.
+ *
+ * Copyright (c) 2022, Intel Corporation.
+ */
+
+#include <sound/soc-acpi.h>
+#include <sound/soc-acpi-intel-match.h>
+
+#define MAX_TPLG_NAME_LEN	64
+
+struct acpi_codec_list {
+	const char *acpi_hid;
+	const char *tplg_name;
+};
+
+static const struct acpi_codec_list codec_list[] = {
+	{
+		.acpi_hid = "10EC5682",
+		.tplg_name = "rt5682",
+	},
+	{
+		.acpi_hid = "RTL5682",
+		.tplg_name = "rt5682",
+	},
+	{
+		.acpi_hid = "10134242",
+		.tplg_name = "cs42l42",
+	},
+	{
+		.acpi_hid = "10508825",
+		.tplg_name = "nau8825",
+	},
+};
+
+static const struct acpi_codec_list amp_list[] = {
+	{
+		.acpi_hid = "RTL1015",
+		.tplg_name = "rt1015",
+	},
+	{
+		.acpi_hid = "RTL1019",
+		.tplg_name = "rt1019",
+	},
+	{
+		.acpi_hid = "MX98357A",
+		.tplg_name = "max98357a",
+	},
+	{
+		.acpi_hid = "MX98360A",
+		.tplg_name = "max98360a",
+	},
+	{
+		.acpi_hid = "MX98373",
+		.tplg_name = "max98373",
+	},
+	{
+		.acpi_hid = "MX98390",
+		.tplg_name = "max98390",
+	},
+	{
+		.acpi_hid = "CSC3541",
+		.tplg_name = "cs35l41",
+	},
+};
+
+static int find_codec_device_and_properties(const struct acpi_codec_list *list,
+					    int list_len,
+					    int *ssp_port)
+{
+	struct acpi_device *adev;
+	struct device *dev;
+	int i, port, ret;
+
+	for (i = 0; i < list_len; i++) {
+		adev = acpi_dev_get_first_match_dev(list[i].acpi_hid, NULL, -1);
+		if (!adev)
+			continue;
+
+		dev = get_device(acpi_get_first_physical_node(adev));
+
+		ret = device_property_read_u32(dev, "intel,ssp-port", &port);
```

Comment:
Where is this property being documented? We really need something like devicetree bindings analogue for the ACPI specific device properties, so we won't reinvent the wheel here and there.